src/matrix/Session.js

```diff
@@ -948,6 +948,23 @@ export class Session {
             return body.room_id;
         });
     }
+
+    waitForRoomFromSync(roomId) {
```

Comment:
Can you integrate this with the `joinRoom` method so that it returns a Promise that eventually resolves to the joined room?

Reply:
True, resolving this in favor of https://github.com/vector-im/hydrogen-web/pull/809#discussion_r928833506